src/azure.js

```diff
@@ -79,13 +79,29 @@ api.declare({
     });
   }
 
-  // Check that the client is authorized to access given account and table
-  if (!req.satisfies({
-    account:    account,
-    table:      tableName,
-    level:      level,
-  })) {
-    return;
+  // We have a complicated scope situation for read-only since we want
+  // read-write to grant read-only permissions as well
+  // req.satisfies will throw an error if it is not satisfied
+  if (level === 'read-only') {
+    try {
+      req.satisfies({
```

Comment:
`req.satisfies` does not throw an exception...

It writes a response and returns `false`, or returns `true`. If called like: `req.satisfies({..}, true)` then it won't write a response...

Thus:
```js
if (!(level == 'read-only' && res.satisfies({account, table, level: 'read-write'}, true) && !res.satisfies({account, table, level})) {
  return;
}
```

should do the trick...

Note, you can also do: `if (!req.satisfies([['scope1'], ['scope2']])) { return; }`, then it's either scope1 or scope2.